Sources/ApolloWebSocket/Compression.swift

```diff
@@ -0,0 +1,183 @@
+//  Created by Joseph Ross on 7/16/14.
+//  Copyright © 2017 Joseph Ross.
+//  Modified by Anthony Miller & Apollo GraphQL on 8/12/21
+//
+//  This is a derived work derived from
+//  Starscream(https://github.com/daltoniam/Starscream)
+//
+//  Original Work License: http://www.apache.org/licenses/LICENSE-2.0
+//  Derived Work License: https://github.com/apollographql/apollo-ios/blob/main/LICENSE
+//
+//  Compression implementation is implemented in conformance with RFC 7692 Compression Extensions
+//  for WebSocket: https://tools.ietf.org/html/rfc7692
+
+import Foundation
+import zlib
+
+class Decompressor {
+  enum Error: Swift.Error {
+    case resetFailed
```

Comment:
What are your thoughts on distinguishing "decompression reset" from "compression reset"? Both errors are simply "reset". Yes within their own class but that also draws question of whether there should be a single error class. `Decompressor.decompressionFailed` and `Compressor.compressionFailed` state the same thing twice.

Reply:
I like scoping the error into the class that throws it. If I used the same error for both, It would have to be an error enum in the global scope. Only reason I did it this way really. @designatednerd opinions? I don't care strongly either way.

Reply:
I personally prefer keeping it scoped to the particular class because then your hands aren't tied by naming of other errors related to other classes. I also am strongly in favor of not rewriting the starscream stuff any more than we absolutely have to.